XenModel/Actions/Updates/DownloadUpdatesXmlAction.cs

```diff
@@ -254,12 +254,14 @@ private void GetXenServerVersions(XmlDocument xdoc)
                     }
 
                     List<XenServerPatch> patches = new List<XenServerPatch>();

```

Comment:
Well, I could, but it wouldn't make any difference to existing functionalities. This is a must for minimalPatches, but not needed for the others.

I think -as part of my bugfix- I am better off not touching what I don't have to and live exactly it as it's been before.

As a future improvement, I'd rather suggest we completely get rid off this code and parse the XML with a DOM parser. (Actually, it has already got some issues like: handling duplicate nodes and not validating the document against a schema)